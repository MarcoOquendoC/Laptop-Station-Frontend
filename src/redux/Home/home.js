import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import api from '../../api/api';

const GET_ITEMS = 'GET_ITEMS';

const initialState = [];

// Thunk
const getItemsInfo = createAsyncThunk(GET_ITEMS, async () => {
  try {
    return await api.fetchItems();
  } catch (error) {
    return error.message;
  }
});

// reducer
export const itemSlice = createSlice({
  name: 'item',
  initialState,
  reducers: { },
  extraReducers: {
    [getItemsInfo.fulfilled]: (state, action) => action.payload,
  },
});

export { getItemsInfo };
export default itemSlice.reducer;
