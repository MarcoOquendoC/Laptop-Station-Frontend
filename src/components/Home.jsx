import React, { useCallback, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { getItemsInfo } from '../redux/Home/home';
import menuImg from '../images/menu.svg';
import logo from '../images/logo_laptop_station.svg';
import arrowLeft from '../images/left.svg';
import arrowRight from '../images/rigth.svg';

const Home = () => {
  const dispatch = useDispatch();

  // Use useCallback to 'memoize' the handleGetItemsInfo function
  const handleGetItemsInfo = useCallback(() => {
    dispatch(getItemsInfo());
  }, [dispatch]);

  useEffect(() => {
    handleGetItemsInfo();
  }, [handleGetItemsInfo]);

  const items = useSelector((store) => store.items);

  return (
    <>
      <div className="principal">
        <div className="header">
          <div className="menu mob"><img src={menuImg} alt="menu" /></div>
          <div className="menu_logo">
            <img className="logo" src={logo} alt="logo" />
          </div>
          <div className="menu mob" />
        </div>

        <nav className="menunav">
          <ul className="menunav_ul">
            <li className="menu_text select"><Link to="/">Home</Link></li>
            <li className="menu_text"><Link to="/laptops">Latops</Link></li>
            <li className="menu_text"><Link to="/reserves">Reserves</Link></li>
            <li className="menu_text admin"><Link to="/addItem">Add item</Link></li>
            <li className="menu_text"><Link to="/logOut">Log Out</Link></li>
          </ul>
        </nav>
      </div>

      <div className="viewfinder">
        <div className="btn left last"><img src={arrowLeft} alt="rigth" /></div>
        <div className="titles">
          <h1>Latest Models</h1>
          <h4>Please select a Laptop Model</h4>
        </div>
        <div className="items">

          {items[0]
            ? items.map((item) => (
              <div key={item.id} className="item">
                <img src={item.image} alt={item.image} />
                <h2>{item.title}</h2>
                <h4>{item.item_model}</h4>
                <p>{item.description}</p>
              </div>
            )) : null}

        </div>
        <div className="btn rigth"><img src={arrowRight} alt="rigth" /></div>
      </div>
    </>
  );
};

export default Home;
