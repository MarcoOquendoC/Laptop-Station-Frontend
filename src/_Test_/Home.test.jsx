import '@testing-library/jest-dom';
import { BrowserRouter as Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import Home from "../components/Home";
import store from "../redux/configureStore";

describe('Testing Home', () => {
  let home;
  beforeAll(async () => {
    home = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Home />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  it('should renders correctly', () => {
    expect(home).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(home).toMatchSnapshot();
  });

  it('renders the component', () => {
    waitFor(() => expect(screen.getByText('Latest Models')).toBeInTheDocument());
    waitFor(() => expect(screen.getByText('Please select a Laptop Model')).toHaveLength(2));
  });

  it('Should have a link to the details page', () => {
    const link = screen.findByTestId('laptop-card');
    waitFor(() => expect(link).toBeInTheDocument());
    waitFor(() => expect(link).toHaveAttribute('href'));
  });
});