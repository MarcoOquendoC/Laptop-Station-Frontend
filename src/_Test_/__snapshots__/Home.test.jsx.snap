// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home should match the snapshot 1`] = `
Array [
  <div
    className="principal"
  >
    <div
      className="header"
    >
      <div
        className="menu mob"
      >
        <img
          alt="menu"
          src="menu.svg"
        />
      </div>
      <div
        className="menu_logo"
      >
        <img
          alt="logo"
          className="logo"
          src="logo_laptop_station.svg"
        />
      </div>
      <div
        className="menu mob"
      />
    </div>
    <nav
      className="menunav"
    >
      <ul
        className="menunav_ul"
      >
        <li
          className="menu_text"
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            Log in
          </a>
        </li>
        <li
          className="menu_text select"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="menu_text"
        >
          <a
            href="/laptops"
            onClick={[Function]}
          >
            Latops
          </a>
        </li>
        <li
          className="menu_text"
        >
          <a
            href="/reservations"
            onClick={[Function]}
          >
            Reserves
          </a>
        </li>
        <li
          className="menu_text admin"
        >
          <a
            className="a__admin"
            href="/addItem"
            onClick={[Function]}
          >
            Add item
          </a>
        </li>
        <li
          className="menu_text"
        />
      </ul>
    </nav>
  </div>,
  <div
    className="viewfinder"
  >
    <div
      className="btn left last"
    >
      <img
        alt="rigth"
        src="left.svg"
      />
    </div>
    <div
      className="titles"
    >
      <h1>
        Latest Models
      </h1>
      <h4>
        Please select a Laptop Model
      </h4>
    </div>
    <div
      className="items"
    />
    <div
      className="btn rigth"
    >
      <img
        alt="rigth"
        src="rigth.svg"
      />
    </div>
  </div>,
]
`;
